### DISCUSSION TITLE
Added .iso with nomodeset for DVD/CD and use with Ventoy. V2023.08.2_25.0_x86-64_0.35

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#215] [Announcements]
I've just added a .iso build [shredos.x86_64-2023.08.2_25.0_x86-64_0.35_nomodeset.iso](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2023.08.2_25.0_x86-64_0.35/shredos.x86_64-2023.08.2_25.0_x86-64_0.35_nomodeset.iso) to the latest version [V2023.08.2_25.0_x86-64_0.35](https://github.com/PartialVolume/shredos.x86_64/releases/tag/v2023.08.2_25.0_x86-64_0.35)

This .iso file can be burned to DVD/CD or copied to a Ventoy USB stick.

### Ventoy Notes
If using with Ventoy you may need to use Ventoy's 'grub2' mode. To enable grub2 mode in Ventoy, when the Ventoy list of Iso/img files appear type control +r to toggle into grub2 mode, then select the ShredOS .iso. The ShredOS iso labeled nomodeset disables the Linux Direct Rendering Manager (DRM) and so remains in simple framebuffer mode. This mode is compatible with most graphics hardware and avoids any buggy graphics drivers.

Also note when booting the .iso with Ventoy you won't see any loading progress, it will just say booting Shredos and you will have to be patient for 30-180 seconds. ShredOS is built with a grub2 module called progress which provides that display however it would appear Ventoy uses it's own grub code? and presumably doesn't include the progress module in their build. If burning the .iso to DVD/CD and booting legacy you will see the progress.

@Binary-Bear @juncture6825 @MinecollYT Can you give this .iso a try and if using with Ventoy please read the notes above regarding grub2 mode. Thanks.

A iso with DRM support is currently being built for those that prefer their text smaller. The DRM version does have some buggy graphics drivers in the current kernel so if you experience any hangs you are better off using the nomodeset .iso

**Comment by Binary-Bear** [comment]
Well done PartialVolume you have done it !!!!

shredos.x86_64-2023.08.2_25.0_x86-64_0.35_nomodeset.iso works for me.  Boot time is roughly 5 minutes but at least shredos boots with Ventoy/Yumi!

**Reply by PartialVolume** [reply]
@Binary-Bear Thanks, I have something else to try in order to solve  your long boot time.

I believe the bios may be providing only a basic v1.1 speed on your system and requires native drivers to be installed in grub for USB. There is a module in grub called nativedisk which I don't currently load. So I'm going to have a play with that and if it boots ok on my systems I'll let you know and if you could try it out and maybe we can bring that boot time right down.

**Comment by PartialVolume** [comment]
Uploaded [shredos-2023.08.2_25.0_x86-64_0.35_20240207_drm_graphics.iso](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2023.08.2_25.0_x86-64_0.35/shredos-2023.08.2_25.0_x86-64_0.35_20240207_drm_graphics.iso) which uses the Linux Display Rendering Manager (DRM) graphics. Warning, this may hang during boot with some graphics hardware. If it does, use the nomodeset .iso instead.

What's the benefit of DRM graphics? It uses the native resolution of the graphics system so the terminal text may appear smaller but avoids line truncation and provides support for sending the computer to sleep and waking, required for unfreezing disk drives for secure erase in hdparm.

**Comment by PartialVolume** [comment]
@Binary-Bear When you get a chance can you boot the iso in Ventoy's normal mode and see if it boots, even if it does take 5 minutes.

It's important that it boots in normal mode as that's when ShredOS's own grub code & grub modules are loaded. It's important that normal mode works otherwise loading nativedisk into ShredOS's code will not work as Ventoy's grub2 mode uses Ventoy's own grub and modules.

If normal mode doesn't work then I'm pretty much stumped as to how to improve the speed on your system. Because grub is using the BIOS firmware to load the USB all that's left is to check that your BIOS is running the latest update for it's firmware.

**Comment by Binary-Bear** [comment]
**shredos.x86_64-2023.08.2_25.0_x86-64_0.35_nomodeset.iso**

Normal Mode
Starts to boot.
Takes roughly 5 minutes.
Ends with uncompression error --system halted

Grub2
Takes roughly 5 minutes.
Boots successfully



**shredos-2023.08.2_25.0_x86-64_0.35_20240207_drm_graphics.iso**

Normal Mode
Starts to boot.
Takes roughly 5 minutes.
Ends with uncompression error --system halted

Grub2
Takes roughly 5 minutes.
Boots successfully

I will be out for the rest of the day today.  Just wanted to send you this first, I am a little late now but shredos comes first! LOL

**Reply by PartialVolume** [reply]
@Binary-Bear I don't know if you have a DVD/CD on that system but the .iso burned to DVD/CD should boot 5 or 6 times faster than USB, at least it does on a old Pentium 4 (32 bit) I have which has the same slow USB speed that you have.

**Comment by Binary-Bear** [comment]
@PartialVolume

I am happy to keep using the usb version as I do not use shredos that frequently and waiting for 5 minutes while it boots isn't so bad.  It would be nice if shredos eventually used the duel boot method you mentioned at some time during its development.

I am more eager to see the new features you are planning for shredos.  I am really keen on the ability to wipe HPA as I feel vulnerable without that option.

I have some feature requests and wondered if you would mind hosting a (sticky) thread in which you could keep editing the first post to show which feature requests you have accepted and those you have rejected?

I would like to ask you to reconsider the current default settings and include a specific wiping method which would help a wonderful project called [veracrypt](https://sourceforge.net/projects/veracrypt/).

Thank you for all the work you do with shredos.

**Comment by PartialVolume** [comment]
>I have some feature requests and wondered if you would mind hosting a (sticky) thread in which you could keep editing the first post to show which feature requests you have accepted and those you have rejected?

Funny you should mention that, but I've been thinking the same thing. I'll put a list together in discussions, it will help me remember what I've said I'll do without having to trawl back through all the issues/posts.

>I would like to ask you to reconsider the current default settings and include a specific wiping method which would help a wonderful project called [veracrypt](https://sourceforge.net/projects/veracrypt/).

I've been asked for a wipe encryption for few times before, presumably this is where the drive itself does the encryption and the key is effectively thrown away? I don't have any issues with adding that however, personally I have some reservations about using drive encryption as an alternative to a traditional wipe. I know it's said it would take the entire existence of the universe to brute force a strong encryption, yet time and time again encryption does get broken, not by brute force but by some undiscovered weakness. So even if I had encrypted a drive I would still wipe it, but maybe that's just me being a bit O.T.T.

>Thank you for all the work you do with shredos.

No problem, glad it's useful.

**Comment by Binary-Bear** [comment]
> Funny you should mention that, but I've been thinking the same thing. I'll put a list together in discussions, it will help me remember what I've said I'll do without having to trawl back through all the issues/posts.

That's great, I will watch the forum waiting to pounce with my list of requests :)

**Comment by Firminator** [comment]
> I've been asked for a wipe encryption for few times before, presumably this is where the drive itself does the encryption and the key is effectively thrown away? I don't have any issues with adding that however, personally I have some reservations about using drive encryption as an alternative to a traditional wipe. I know it's said it would take the entire existence of the universe to brute force a strong encryption, yet time and time again encryption does get broken, not by brute force but by some undiscovered weakness. So even if I had encrypted a drive I would still wipe it, but maybe that's just me being a bit O.T.T.

No, that's a valid concern. Encryption is a huge complex codebase and the multiple different implementations are under constant attack/research leading to multiple occasions per year of published vulnerabilities. Also - I might tap into conspiracy theories here - I remember how it was a mystery that the developer of TrueCrypt suddenly disappeared and the code became stale... that's when VeraCrypt picked up.

Maybe add it as an alterbative for PRNG wiping. Or create a custom wipe with zeroing out, then encrypt or some combination of it like 1,0,Encrypt? I kind of recall that the NIST guidelines might have a paragraph about this topic, but not 100% sure.

**Reply by BigPanda97** [reply]
> Maybe add it as an alterbative for PRNG wiping. Or create a custom wipe with zeroing out, then encrypt or some combination of it like 1,0,Encrypt?

That's bullshit. What should be the advantage here? Encrypting the data afterwards is the same as writing PRNG data on it, there is no benefit. Crypto-Erase only works when ALL data on the hard drive was encrypted since the creation of the data itself so that the data was never stored unencrypted/decrypted on the drive. However, some governments like Germany recommend using a solid cipher like AES as a PRNG (in CTR mode) over other generators like ISAAC because of the better statistical quality of the pseudo-randomness from a cipher.

**Reply by Firminator** [reply]
There is no advantage... unless folks don't trust PRNG (i.e. [compromised RNG](https://www.imperialviolet.org/2015/12/19/juniper.html)). The context was folks pushing or insisting on implementing encryption as a wiping method and it crossed my mind that it could be added as an alternative... althoug a code-expensive alternative increasing the attack surface.
Well it's all a moot point now anyways as Binary-Bear clarified his "request does not involve encrypting the drive."

**Comment by Binary-Bear** [comment]
My request does not involve encrypting the drive.  It would not make sense to encrypt the drive when our intention is to delete the data.

I have not posted the specifics yet because I was waiting for the new requests thread to be opened.
